src/Nethermind/Nethermind.Consensus/Validators/BlockValidator.cs

```diff
@@ -193,31 +204,53 @@ private bool ValidateWithdrawals(Block block, IReleaseSpec spec, out string? err
         return true;
     }
 
-    private bool ValidateBlobs(Block block, IReleaseSpec spec, out string? error)
+    private bool ValidateTransactionsDataGas(Block block, IReleaseSpec spec, out string? error)
     {
-        if (spec.IsEip4844Enabled && block.ExcessDataGas is null)
+        if (!spec.IsEip4844Enabled)
         {
-            error = "ExcessDataGas field is not set.";
-            if (_logger.IsWarn) _logger.Warn(error);
-            return false;
+            error = null;
+            return true;
         }
 
-        if (!spec.IsEip4844Enabled && block.ExcessDataGas is not null)
+        int blobsInBlock = 0;
+        UInt256 dataGasPrice = UInt256.Zero;
+        for (int txIndex = 0; txIndex < block.Transactions.Length; txIndex++)
         {
-            error = "ExcessDataGas field should not have value.";
-            if (_logger.IsWarn) _logger.Warn(error);
-            return false;
+            Transaction transaction = block.Transactions[txIndex];
```

Comment:
A minor thing to consider: We have to loop through all transactions when we're calling TxValidator. Shouldn't we move this validation to the already existing loop?

Reply:
Good catch, let me shuffle it a bit